src/views/Header/Header.js

```diff
@@ -327,7 +330,13 @@ class Header extends React.PureComponent<Props, State> {
 
     const { options } = this.props.getScreenDetails(scene);
     const { headerStyle } = options;
-    const appBarHeight = Platform.OS === 'ios' ? (isLandscape ? 32 : 44) : 56;
+    const viewHeight =
+      typeof rest.layout.height === 'number' ? rest.layout.height : null;
+    const isHeightConstrained = viewHeight ? viewHeight < 500 : !isTablet;
```

Comment:
500 is a heuristic; most phones in landscape are less then 500 and most tablets are greater. But more importantly it is the threshold I've chosen for determining if their is a height constraint, and if we need to shrink the header to make space for more content (similar to Apple's notion of size classes). 

Should I use a smaller value like 300, 350, or 400? The key is to make it show that on iPhone, the small header only shows on iPhone in landscape (not iPad in tablet).

Reply:
why are we using a number at all? we know if the device is an ipad or an iphone, and we know if the device is landscape or portrait, that seems to be all of the information that we need.

Reply:
That's a good point, I was told in #3041 by @satya164 that I should avoid using an isPad constant, but I think that's because neither of us knew about `Platform.isPad`.  I'll revise and push a new version.